src/package.rs

```diff
@@ -339,6 +358,33 @@ impl Package {
         Ok(())
     }
 
+    async fn add_rust<W: std::io::Write>(
+        &self,
+        progress: &impl Progress,
+        archive: &mut Builder<W>,
+    ) -> Result<()> {
+        match &self.source {
+            PackageSource::Local {
+                rust: Some(rust_pkg),
+                ..
+            } => {
+                progress.set_message("adding rust binaries");
+                let dst = match self.output {
+                    PackageOutput::Zone { .. } => {
+                        let dst = Path::new("/opt/oxide").join(&self.service_name).join("bin");
+                        add_directory_and_parents(archive, &dst)?;
+                        archive_path(&dst)?
+                    }
+                    PackageOutput::Tarball => PathBuf::from(""),
+                };
+                rust_pkg.add_binaries_to_archive(archive, &dst)?;
+                progress.increment(1);
+            }
+            _ => {}
```

Comment:
Would it be slightly clearer / more future-proof if `PackageSource` had getter methods like `.rust_package() -> Option<RustPackage>` (for use here) and `.blobs() -> Option<&[PathBuf]>` method (for use in `add_blobs()` below)? Two thoughts:

1. It would make these methods slightly cleaner since you wouldn't need to `match` (and readers wouldn't have to think about whether `_` is still applicable)
2. The matches themselves would be more tightly coupled with the `PackageSource` type, which would probably make it more likely that anyone updating `PackageSource` itself would notice if they needed to change them

Reply:
Sure thing, added methods for both.